File: backend/src/main/java/com/dayglyph/service/AuthService.java
package com.dayglyph.service;

import com.dayglyph.dto.request.LoginRequest;
import com.dayglyph.dto.request.RegisterRequest;
import com.dayglyph.dto.response.AuthResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

}

File: frontend/src/services/authService.ts
import api from "./api";

interface RegisterRequest {
  name: string;
  email: string;
  password: string;
}

interface LoginRequest {
  email: string;
  password: string;
}

export async function register(request: RegisterRequest) {

  const response = await api.post(
    "/auth/register",
    request
  );

  return response.data.data;
}

export async function login(request: LoginRequest) {

  const response = await api.post(
    "/auth/login",
    request
  );

  return response.data.data;
}